operantanalysis/operantanalysis.py

```diff
@@ -1,108 +1,158 @@
-
-
-def leverpressing(eventcode, lever1, lever2=False):
-    lever1presses = eventcode.count(lever1)
-    if bool(lever2):
-        lever2presses = eventcode.count(lever2)
+    cue_on = [i for i, event in enumerate(eventcode) if event == code_on]
+    cue_off = [i for i, event in enumerate(eventcode) if event == code_off]
+    iti_on = [i for i, event in enumerate(eventcode) if event == code_off or event == 'StartSession']
+    all_poke_rpm = []
+    all_poke_iti_rpm = []
+
+    for i in range(len(cue_on)):
+        cue_on_idx = cue_on[i]
+        cue_off_idx = cue_off[i]
+        iti_on_idx = iti_on[i]
+        cue_length_sec = (timecode[cue_off_idx] - timecode[cue_on_idx])
+        poke_rpm = ((eventcode[cue_on_idx:cue_off_idx].count('PokeOn1')) / (cue_length_sec / 60))
+        all_poke_rpm += [poke_rpm]
+        iti_poke = 0
+        for x in range(iti_on_idx, cue_on_idx):
+            if eventcode[x] == 'PokeOn1' and timecode[x] >= (timecode[cue_on_idx] - cue_length_sec):
+                iti_poke += 1
+        iti_poke_rpm = iti_poke / (cue_length_sec / 60)
+        all_poke_iti_rpm += [iti_poke_rpm]
+
+    return round(statistics.mean(all_poke_rpm), 3), round(statistics.mean(all_poke_iti_rpm), 3)
+
+
+def lever_pressing(eventcode, lever1, lever2=False):
+    """
+
+    :param eventcode: list of event codes from operant conditioning file
+    :param lever1: eventcode for lever pressing
+    :param lever2: optional parameter for second lever eventcode if two levers are used
+    :return: count of first lever presses, second lever presses, and total lever presses, as int
+    """
+    lever1_presses = eventcode.count(lever1)
+    if lever2:
+        lever2_presses = eventcode.count(lever2)
     else:
-        lever2presses = 0
-    totalleverpresses = lever1presses + lever2presses
+        lever2_presses = 0
+    total_lever_presses = lever1_presses + lever2_presses
 
-    return lever1presses, lever2presses, totalleverpresses
+    return lever1_presses, lever2_presses, total_lever_presses
+
+
+def lever_press_latency(timecode, eventcode, lever_on, lever_press):
+    """
+
+    :param timecode: list of times (in seconds) when events occurred
+    :param eventcode: list of events that happened in a session
+    :param leveron: event name for lever presentation
+    :param leverpress: event name for lever press
+    :return: the mean latency to press the lever in seconds
+    """
+    lever_on = [i for i, event in enumerate(eventcode) if event == lever_on or event == 'EndSession']
+    press_latency = []
+    for i in range(len(lever_on) - 1):
+        lever_on_idx = lever_on[i]
+        if lever_press in eventcode[lever_on_idx:lever_on[i + 1]]:
+            lever_press_idx = eventcode[lever_on_idx:lever_on[i + 1]].index(lever_press)
+            press_latency += [round(timecode[lever_on_idx + lever_press_idx] - timecode[lever_on_idx], 2)]
+            break
+        else:
+            None
```

Comment:
kinda not needed `else:` then.  If you really need to "do nothing", using `pass` statement would be more conventional.

Reply:
I'll change to the pass statement!